lib/puppet.rb

```diff
@@ -251,9 +251,11 @@ def self.base_context(settings)
       # doesn't exist
       default_environment = Puppet[:environment].to_sym
       if default_environment == :production
+        modulepath = settings[:modulepath]
+        modulepath = modulepath.nil? ? basemodulepath : Puppet::Node::Environment.split_path(modulepath)
```

Comment:
The default `Puppet[:modulepath]` is [an empty string](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L1071):

```
(byebug) Puppet[:modulepath]
""
```